airflow/models/taskmixin.py

```diff
@@ -140,6 +142,27 @@ def set_setup_teardown_ctx_dependencies(self, other: DependencyMixin | Sequence[
             return
         SetupTeardownContext.update_context_map(other)
 
+    def set_taskgroup_ctx_dependencies(self, other: DependencyMixin | Sequence[DependencyMixin]):
+        from airflow.utils.task_group import TaskGroupContext
+
+        if not TaskGroupContext.active:
+            return
+        from airflow.models.xcom_arg import PlainXComArg
+
+        op1 = self
+        if isinstance(self, PlainXComArg):
+            op1 = self.operator
+        TaskGroupContext.add_task(op1)
+        if isinstance(other, Sequence):
+            for op in other:
```

Comment:
This kind of iterating is also present in `XComArg.iter_xcom_references()`, we may want to reuse it or somewhat refactor the logic.

Reply:
Since `other` could also be a classic operator or a list of it, I don't think reusing `XComArg.iter_xcom_references()` will work here?

Reply:
`iter_xcom_references` is a static method that takes anything, not an instance method acting on XComArg.

https://github.com/apache/airflow/blob/21e8f878a3c91250d0d198c6c3675b4b350fcb61/airflow/models/xcom_arg.py#L99-L104

Reply:
I added a method in DependencyMixin instead of a refactor to `iter_xom_references`. Using `iter_xcom_references` I had issues where we checked if it was an instance of `AbstractOperator` and iterated through the template fields. Instead of refactoring it, I opted to add another method on DependencyMixn to just return operator